apps/wallet-mobile/src/features/Swap/useCases/StartSwapScreen/CreateOrder/CreateOrder.tsx

```diff
@@ -239,6 +251,77 @@ export const CreateOrder = () => {
 
 const Actions = ({style, ...props}: ViewProps) => <View style={[styles.actions, style]} {...props} />
 
+const useSellError = (errors: Array<string> = []): string => {
+  const noPoolError = useNoPoolError()
+  const notEnoughBalanceError = useNotEnoughBalanceError()
+
+  const allErrors = [noPoolError, notEnoughBalanceError, ...errors]
+  const sellError = allErrors.find((error) => !isEmptyString(error))
+
+  return sellError ?? ''
+}
+
+const useBuyError = (errors: Array<string> = []): string => {
+  const noPoolError = useNoPoolError()
+  const notEnoughSupplyError = useNotEnoughSupplyError()
+
+  const allErrors = [noPoolError, notEnoughSupplyError, ...errors]
+  const buyError = allErrors.find((error) => !isEmptyString(error))
+
+  return buyError ?? ''
+}
+
+const useNotEnoughSupplyError = (): string => {
+  const strings = useStrings()
+  const {orderData} = useSwap()
+  const {isBuyTouched, isSellTouched} = useSwapTouched()
+  const pool = orderData.selectedPoolCalculation?.pool
+  const {tokenId, quantity} = orderData.amounts.buy
+  const poolSupply = tokenId === pool?.tokenA.tokenId ? pool?.tokenA.quantity : pool?.tokenB.quantity
+  const hasSupply = !Quantities.isGreaterThan(quantity, poolSupply ?? Quantities.zero)
+
+  const notEnoughSupplyError =
+    (!Quantities.isZero(quantity) && !hasSupply) || (isSellTouched && isBuyTouched && pool === undefined)
+      ? strings.notEnoughSupply
+      : ''
+
+  return notEnoughSupplyError
+}
+
+const useNotEnoughBalanceError = (): string => {
+  const strings = useStrings()
+  const {orderData} = useSwap()
+  const {isBuyTouched} = useSwapTouched()
+  const wallet = useSelectedWallet()
+  const {tokenId, quantity} = orderData.amounts.sell
+  const balance = useBalance({wallet, tokenId})
+
+  const hasBalance = !Quantities.isGreaterThan(
+    Quantities.sum([
+      tokenId === wallet.primaryTokenInfo.id ? orderData.amounts.sell.quantity : Quantities.zero,
+      orderData.selectedPoolCalculation?.cost.batcherFee.quantity ?? Quantities.zero,
+      orderData.selectedPoolCalculation?.cost.frontendFeeInfo.fee.quantity ?? Quantities.zero,
+      orderData.selectedPoolCalculation?.pool.deposit.quantity ?? Quantities.zero,
+    ]),
+    balance,
+  )
```

Comment:
Maybe here we need to separate primary token amount and other token amount right?

twoo types of balance: primare token balance + other types

Reply:
@banklesss if you need all fees summed up, add them to makeOrderCalculations in another `cost` field, so you don't have to sum them in the app's front

Reply:
@stackchain we need to clarify if the frontendfee should be added or not
We said let's calculate it so we know how much it is, but we don't apply it
If we're not applying it it shouldn't be summed up here

Reply:
`ptTotalFee` added to `orderData.selectedPoolCalculation?.cost`

Reply:
the same way we have the other feeds, let's have the ptTotalFeeWithFeF and without

Reply:
and the client decides